src/main.cpp

```diff
@@ -6162,35 +6202,23 @@ bool static ProcessMessage(CNode* pfrom, string strCommand, CDataStream& vRecv,
 
         int count=0;
         size_t i;
-        if (pfrom->xThinBlockHashes.size() != pfrom->thinBlock.vtx.size())  // Because the next loop assumes this
-          {
-            LogPrint("thin", "Inconsistent thin block data.  Aborting the thin block\n");
+        for (i = 0; i < pfrom->thinBlock.vtx.size(); i++)
+        {
+            if (pfrom->thinBlock.vtx[i].IsNull())
             {
-                LOCK(cs_vNodes);
-                pfrom->mapThinBlocksInFlight.erase(inv.hash);
-                pfrom->thinBlockWaitingForTxns = -1;
-                pfrom->thinBlock.SetNull();
+	        std::map<uint64_t, CTransaction>::iterator val = mapMissingTx.find(pfrom->xThinBlockHashes[i]);
```

Comment:
May I suggest using `auto` ?

```
auto val =
```

Leaves you with more space to write longer and more useful variable names than `val` ;)